### PR TITLE
Support roundingType and outputSizes for pool2d on OV/DML

### PR DESCRIPTION
**Pull request by mingmingtasd**
PTAL, thanks! @fujunwei @huningxin  I also raised an issue #184 on DML for some un-supported cases.

### CODE REVIEW

File: webnn.json [hunk 1]
```diff
@@ -345,7 +352,10 @@
       {"name": "dilations count", "type": "uint32_t", "default": 0},
       {"name": "dilations", "type": "int32_t", "annotation": "const*", "length": "dilations count", "optional": true},
       {"name": "auto pad", "type": "auto pad", "default": "explicit"},
-      {"name": "layout", "type": "input operand layout", "default": "nchw"}
+      {"name": "layout", "type": "input operand layout", "default": "nchw"},
+      {"name": "rounding type", "type": "rounding type", "default": "floor"},
+      {"name": "output sizes count", "type": "uint32_t", "default": 0},
```
Comment (fujunwei) [inline webnn.json:357]: @huningxin Why does spec define the outputSizes that can be inferred with input arguments?

File: src/webnn_native/ops/Pool2d.cpp [hunk 2]
```diff
@@ -96,13 +102,44 @@ namespace webnn_native { namespace op {
...
+        int32_t outputHeight, outputWidth;
+        int32_t floorOutputHeight =
             1 + (inputHeight - windowHeight + paddingBeginningHeight + paddingEndingHeight) /
                     mStride[0];
-        int32_t outputWidth =
+        int32_t floorOutputWidth =
             1 +
             (inputWidth - windowWidth + paddingBeginningWidth + paddingEndingWidth) / mStride[1];
+        int32_t ceilOutputHeight = ceil(
+            1 + 1.0 * (inputHeight - windowHeight + paddingBeginningHeight + paddingEndingHeight) /
```
[the remark this answers is not in the document]
Comment (mingmingtasd) [reply src/webnn_native/ops/Pool2d.cpp:115]: `1.0 *` will affect the `(inputHeight - windowHeight + paddingBeginningHeight + paddingEndingHeight) `at first, it's not easy to extract.

File: src/webnn_native/ops/Pool2d.cpp [hunk 3]
```diff
@@ -96,13 +102,44 @@ namespace webnn_native { namespace op {
...
+                    mStride[0]);
+        int32_t ceilOutputWidth =
+            ceil(1 + 1.0 * (inputWidth - windowWidth + paddingBeginningWidth + paddingEndingWidth) /
+                         mStride[1]);
+        if (mOptions.outputSizes == nullptr) {
+            if (mOptions.roundingType == ml::RoundingType::Floor) {
+                outputHeight = floorOutputHeight;
```
[the remark this answers is not in the document]
Comment (mingmingtasd) [reply src/webnn_native/ops/Pool2d.cpp:122]: Agree, thanks! @fujunwei

**Comment by mingmingtasd** [comment]
Updated,PTAL, thanks! @fujunwei

File: src/webnn_native/ops/Pool2d.cpp [hunk 4]
```diff
@@ -96,13 +102,43 @@ namespace webnn_native { namespace op {
...
+                                                                            : ceilOutputHeight;
+            outputWidth = mOptions.roundingType == ml::RoundingType::Floor ? floorOutputWidth
+                                                                           : ceilOutputWidth;
+        } else {
+            outputHeight = mOptions.outputSizes[0];
+            outputWidth = mOptions.outputSizes[1];
+            // The specified outputSizes should match either floor or ceil rounding type.
+            if (!((outputHeight == floorOutputHeight && outputWidth == floorOutputWidth) ||
+                  (outputHeight == ceilOutputHeight && outputWidth == ceilOutputWidth))) {
+                return DAWN_VALIDATION_ERROR("Invalid output sizes.");
```
[the remark this answers is not in the document]
Comment (mingmingtasd) [reply src/webnn_native/ops/Pool2d.cpp]: Good suggestion, updated yet, PTAL, thanks! @fujunwei

**Comment by fujunwei** [comment]
Merging, thanks.

**Pull request merged**